perceval/utils/parameter.py

```diff
@@ -220,186 +216,100 @@ def max(self, m: bool):
+            raise RuntimeError(f"Missing parameters: {free_symbol_names - pcvl_param_names}")
 
-    def __add__(self, other):
-        if isinstance(other, Expression):
-            return Expression(f"({self.name}+{other.name})", other._params | self._params)
-        elif isinstance(other, (Parameter, P)):
-            return Expression(f"({self.name}+{other.name})", {other} | self._params)
-        elif isinstance(other, (int, float)):
-            return Expression(f"({self.name}+{other})", self._params)
-        raise TypeError("Unsupported parameter operation.")
-
-    def __radd__(self, other):
-        return self.__add__(other)
-
-    def __sub__(self, other):
-        if isinstance(other, Expression):
-            return Expression(f"({self.name}-{other.name})", other._params | self._params)
-        elif isinstance(other, Parameter):
-            return Expression(f"({self.name}-{other.name})", {other} | self._params)
-        elif isinstance(other, (int, float)):
-            return Expression(f"({self.name}-{other})", self._params)
-        raise TypeError("Unsupported parameter operation.")
+    def __repr__(self):
+        return f"Expression({self.name[1:-1]}, parameters={self._params})"
 
-    def __rsub__(self, other):
-        if isinstance(other, (int, float)):
-            return Expression(f"({other}-{self.name})", self._params)
-        raise TypeError("Unsupported operation.")
-    
-    def __truediv__(self, other):
-        if isinstance(other, Expression):
-            return Expression(f"({self.name}/{other.name})", other._params | self._params)
-        elif isinstance(other, Parameter):
-            return Expression(f"({self.name}/{other.name})", {other} | self._params)
-        elif isinstance(other, (int, float)):
-            return Expression(f"({self.name}/{other})", self._params)
-        raise TypeError("Unsupported parameter operation.")
-    
-    def __pow__(self, other):
-        if isinstance(other, (int, float)):
-            return Expression(f"{self.name}^{other}", self._params)
-        raise TypeError("Unsupported operation.")
-    
-    def _create_missing_parameters(self) -> set:
-        r"""Creates parameters not specified in Expression name"""
-        param_names = re.findall(r'\b[a-zA-Z_][a-zA-Z0-9_]*\b', self.name)
-        for name in param_names:
-            if name not in [p.name for p in self._params]:
-                self._params.add(P(name))
-        
-    def _update_value(self):
+    def __float__(self):
         """Updates Expression with respect to any changes made to parent Parameters"""
         if any(not param.defined for param in self._params):
-            self._value = None
-        else:
-            self._value = sp.S(self.name).subs({param.name : param._value for param in self._params})
+            raise ValueError("Expression is symbolic, cannot compute its numerical value")
+        return float(self.spv.subs({param.name : param._value for param in self._params}))
```

Comment:
Only the basic Parameters are used here.
I'm not sure that we want to keep the Expressions in `self._params`

Reply:
Not sure I understand your comment. Only `Parameter` are stored in `self._params`

Reply:
You're right, I thought it was still the case but you changed it